sdk/rust/oak/src/lib.rs

```diff
@@ -42,15 +43,15 @@ pub type Handle = u64;
 /// Wrapper for a handle to the read half of a channel.
 ///
 /// For use when the underlying [`Handle`] is known to be for a receive half.
-#[derive(Debug, Copy, Clone, PartialEq)]
+#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
```

Comment:
I think this is a bad idea; if you make it possible to serialize handles, people will be encouraged to serialize handles and then be confused when the deserialized handle doesn't work (because it has crossed a Node boundary and so is no longer valid).

Reply:
Good point, here I added these in order to be able to use `serde` for most of the serialisation, and then manually pull out the handles and send them separately. Something like [FIDL](https://fuchsia.dev/fuchsia-src/development/languages/fidl/reference/wire-format) may replace (in fact, combine) the serialisation and sending steps. I will add a comment to clarify this.

Reply:
Added some comments (and also on `Handle`).

Reply:
I still think this is a big footgun, so maybe add an issue to go with the TODO?

Reply:
Sure, though I think the main issue is that `Handle` itself is just a u64 and therefore already trivially serializable (even if we don't specify `derive(Serialize)` on that type).

Created https://github.com/project-oak/oak/issues/427 to track and updated TODOs.

Reply:
Ref for tracking: #429 will implement opaque & non-serializable `Handle`s